hibernate-jfr/hibernate-jfr.gradle

```diff
@@ -0,0 +1,27 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+description = 'Integration for JDK JFR into Hibernate O/RM'
+
+apply from: rootProject.file( 'gradle/published-java-module.gradle' )
+
+dependencies {
+    implementation project( ':hibernate-core' )
+
+    testImplementation project( ':hibernate-testing' )
+    testImplementation testLibs.jfrUnit
+}
+
+
+// JfrUnit requires JDJ 17
+compileTestJava.onlyIf{
+    jdkVersions.test.release.asInt() >= 17 && jdkVersions.explicit
```

Comment:
@beikov 
Was testing this while working in pair with @dreab8 , this condition wasn't working on my IDEA as we have the toolset forcing the project to test with JDK11.
I suspect that even chancing the `jdkVersions.explicit` part, we'd be in trouble during a release as the global build would be forced to test with JDK11 - we'd need to change the toolchain use so to allow this particular module to request running tests with JDK17.

```
test {
    //Testing JFR events require JDK > 16
    javaLauncher = javaToolchains.launcherFor {
        languageVersion = JavaLanguageVersion.of(17)
    }
}
```

However now I had to enable auto-provisioning in `./gradle.properties` :
```
org.gradle.java.installations.auto-detect=true
```

Not sure if that's going to be a problem..

Reply:
I don't think the use of the `jdkVersions.explicit` property is correct/needed, but I would rather prefer not to do automatic java installations if possible. Tests should only run if a Java 17 runtime is used.

Reply:
removed `jdkVersions.explicit`

Reply:
`jdkVersions` is OUR object.  `.explicit` simply indicates whether these values were explicitly set.  It is not used in any way for automatic installation (which we do not do at all)